Add support for VirtualSystemCollection

bostko opened the pull request:
- added a test for an upgraded envelope.xml

bostko committed:
Add support for VirtualSystemCollection

- added a test for an upgraded envelope.xml

bostko commented:
Note that the tests for the envelope_upgraded.xml are failing :(

aledsage commented on src/main/java/org/jclouds/vcloud/director/v1_5/domain/dmtf/Envelope.java:
```diff
@@ -81,6 +85,6 @@ public ReferencesType getReferences() {
 
    public VirtualSystem getVirtualSystem() {
-      return virtualSystem;
+      return virtualSystem != null ? virtualSystem : virtualSystemCollection.getVirtualSystems().iterator().next();
```
Should we have another method that returns all the `VirtualSystem`s?

Maybe we should be more defensive - for example if `virtualSystemCollection` is also null or we got back an empty VirtualSystemCollection then this would give a `NoSuchElementException`. vCD probably won't do that, but we other areas have seen different things returned in different environments that is impossible to guess in advanced by just looking at the docs.

What do we want to return if there are none? We could do:

```
if (virtualStream != null) {
    return virtualSystem;
} else if (virtualSystemCollection == null || virtualSystemCollection.getVirtualSystems() == null) {
    return null;
} else {
    return Iterables.getFirst(virtualSystemCollection.getVirtualSystems(), null);
}
```

andreaturli replied:
@aledsage +1
also I'm not totally sure we should get the first VirtualSystem. Do they all contain the same types? or we can filter more accurately?

aledsage commented:
@andreaturli can you take a look as well please?

andreaturli commented on src/test/java/org/jclouds/vcloud/director/v1_5/domain/dmtf/EnvelopeTest.java:
```diff
@@ -44,4 +44,11 @@ public void testUnmarshallEnvelope() throws JAXBException {
+      unmarshaller.setEventHandler(new javax.xml.bind.helpers.DefaultValidationEventHandler());
+      File xml = new File("src/test/resources/dmtf/envelope_upgraded.xml");
+      Envelope envelope = (Envelope) unmarshaller.unmarshal(xml);
+      assertNotNull(envelope);
```
could you add more assertions here, please?

andreaturli commented on src/test/resources/dmtf/envelope_upgraded.xml:
```diff
@@ -0,0 +1,193 @@
+    <vcloud:NetworkConfig networkName="vAppNet-Deployment_Network_001">
+        <vcloud:Description/>
+        <vcloud:Configuration>
+            <vcloud:IpScope>
```
seems that it used to be `IpScopes` but now it is missing. Which vCD version is producing this one?

bostko committed:
Add support for IpScope directly in NetworkConfiguration node

 Make parsable other nodes: AnnotationSection, StartupSection

bostko commented:
@andreaturli @aledsage  I added `IpScope`. However I think no information is retrieved from it.
It is not in advanced networking as well.

I added support for more XmlNodes so the other test could pass.

I will try to add some assertions in the next hours.

bostko committed:
Added osType test

andreaturli commented:
close as we need to re-open it against `1.9.x` branch
